renderer/src/components/Totals/Totals.tsx

```diff
@@ -65,6 +78,10 @@ const Totals = ({ selectedDate }) => {
     };
   }, [selectedDate, period]);
 
+  useEffect(() => {
+    changePeriod(period.periodName, selectedDate);
+  }, [selectedDate]);
```

Comment:
I guess you should add the period as dependencies for useEffect.

Reply:
This causes an infinite recursion. The period is overwritten, the useEffect is triggered, the period is overwritten...